mypy/checkexpr.py

```diff
@@ -645,15 +649,48 @@ def check_call_expr_with_callee_type(self,
                                          callee_type: Type,
                                          e: CallExpr,
                                          callable_name: Optional[str],
-                                         object_type: Optional[Type]) -> Type:
+                                         object_type: Optional[Type],
+                                         member: Optional[str] = None) -> Type:
         """Type check call expression.
 
         The given callee type overrides the type of the callee
         expression.
+
+        The 'callable_name' and 'object_type' are used to call plugin hooks.
+        If 'callable_name' is None but 'member' is not None (member call), try constructing
+        'callable_name' using 'object_type' (the base type on which the method is called),
+        for example 'typing.Mapping.get'.
         """
-        # Try to refine the call signature using plugin hooks before checking the call.
-        callee_type = self.transform_callee_type(
-            callable_name, callee_type, e.args, e.arg_kinds, e, e.arg_names, object_type)
+        if callable_name is None and member is not None:
+            assert object_type is not None
+            callable_name = self.method_fullname(object_type, member)
+        if callable_name:
+            # Try to refine the call signature using plugin hooks before checking the call.
+            callee_type = self.transform_callee_type(
+                callable_name, callee_type, e.args, e.arg_kinds, e, e.arg_names, object_type)
+        # Unions are special-cased to allow plugins to act on each item in the union.
+        elif member is not None and isinstance(object_type, UnionType):
+            res = []  # type: List[Type]
+            for typ in object_type.items:
+                # Member access errors are already reported when visiting the member expression.
+                self.msg.disable_errors()
+                item = analyze_member_access(member, typ, e, False, False, False,
+                                             self.msg, original_type=object_type, chk=self.chk,
+                                             in_literal_context=self.is_literal_context())
+                self.msg.enable_errors()
+                item = self.narrow_type_from_binder(e.callee, item)
+                # TODO: This check is not one-to-one with non-special-cased behavior.
```

Comment:
Do you know what issues this may cause? Currently it's not clear if this is a serious problem or not.

Reply:
Actually I found a cleaner solution, I will push an update soon.